subprojects/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/transform/ArtifactTransformIntegrationTest.groovy

```diff
@@ -1421,7 +1421,7 @@ Found the following transforms:
         then:
         failure.assertHasDescription("Execution failed for task ':resolve'.")
         failure.assertHasCause("Could not resolve all files for configuration ':compile'.")
-        failure.assertHasCause("Failed to transform file 'a.jar' to match attributes {artifactType=size} using transform ToNullTransform")
+        failure.assertHasCause("Failed to transform file 'a.jar' to match attributes {artifactType=size}")
```

Comment:
Can we add some assertion about the remaining stack trace?

IIUC, the original failure will be wrapped in an execution exception as well. Do we want to have both wrappings or should we only have one?

Reply:
I guess we should only ever wrap the cause of an `ExecutionException` in an `ArtifactTransformException`, right? I.e. if there's any other exception, we should just throw that upwards (and ultimately fail the build), since we don't know how to handle it.

Reply:
Well, there are also the exceptions caused by resolving artifacts and that is not part of the transformation. I guess we don't see those in immediate transformations, right?

Reply:
> IIUC, the original failure will be wrapped in an execution exception as well. Do we want to have both wrappings or should we only have one?

I think this is a different topic. We should probably unwrap the `ExecutionException`, but maybe we should use a `Try` instead of an `ExecutionException`, too. Let's create a separate issue about this.